src/Admin/AmpPlugins.php

```diff
@@ -61,6 +62,10 @@ public static function get_registration_action() {
 	 */
 	public static function is_needed() {
 
+		if ( ! Services::get( 'dependency_support' )->has_support() ) {
+			return false;
+		}
```

Comment:
The inclusion of this and the same code in `AmpThemes::is_needed()` is causing fatal errors:

```
PHP Fatal error:  Uncaught AmpProject\AmpWP\Exception\InvalidService: The service ID "dependency_support" is not recognized and cannot be retrieved. in /app/public/wp-content/plugins/amp/src/Exception/InvalidService.php:57
Stack trace:
#0 /app/public/wp-content/plugins/amp/src/Infrastructure/ServiceContainer/SimpleServiceContainer.php(39): AmpProject\AmpWP\Exception\InvalidService::from_service_id('dependency_supp...')
#1 /app/public/wp-content/plugins/amp/src/Services.php(55): AmpProject\AmpWP\Infrastructure\ServiceContainer\SimpleServiceContainer->get('dependency_supp...')
#2 /app/public/wp-content/plugins/amp/src/Admin/AmpPlugins.php(65): AmpProject\AmpWP\Services::get('dependency_supp...')
#3 /app/public/wp-content/plugins/amp/src/Infrastructure/ServiceBasedPlugin.php(413): AmpProject\AmpWP\Admin\AmpPlugins::is_needed()
#4 /app/public/wp-content/plugins/amp/src/Infrastructure/ServiceBasedPlugin.php(385): AmpProject\AmpWP\Infrastructure\ServiceBasedPlugin->maybe_register_service('admin.amp_plugi...', 'AmpProject\\AmpW...')
 in /app/public/wp-content/plugins/amp/src/Exception/InvalidService.php on line 57
```

@schlessera Any idea why this would be? I thought this would have been fixed by https://github.com/ampproject/amp-wp/pull/6484.

Reply:
I can fix this by making `dependency_support` support appear first in the `SERVICES` list:

```diff
diff --git a/src/AmpWpPlugin.php b/src/AmpWpPlugin.php
index 01575d06e..197f0e82d 100644
--- a/src/AmpWpPlugin.php
+++ b/src/AmpWpPlugin.php
@@ -69,6 +69,7 @@ final class AmpWpPlugin extends ServiceBasedPlugin {
 	 * @var string[]
 	 */
 	const SERVICES = [
+		'dependency_support'                 => DependencySupport::class,
 		'admin.analytics_menu'               => Admin\AnalyticsOptionsSubmenu::class,
 		'admin.google_fonts'                 => Admin\GoogleFonts::class,
 		'admin.onboarding_menu'              => Admin\OnboardingWizardSubmenu::class,
@@ -91,7 +92,6 @@ final class AmpWpPlugin extends ServiceBasedPlugin {
 		'cli.validation_command'             => Cli\ValidationCommand::class,
 		'css_transient_cache.ajax_handler'   => Admin\ReenableCssTransientCachingAjaxAction::class,
 		'css_transient_cache.monitor'        => BackgroundTask\MonitorCssTransientCaching::class,
-		'dependency_support'                 => DependencySupport::class,
 		'dev_tools.block_sources'            => DevTools\BlockSources::class,
 		'dev_tools.callback_reflection'      => DevTools\CallbackReflection::class,
 		'dev_tools.error_page'               => DevTools\ErrorPage::class,
```

Reply:
There are other services with `is_needed` methods that are calling `Services::get( 'dependency_support' )->has_support()`:

* `PluginActivationSiteScan`
* `ValidationCounts`
* `Polyfills`
* `PairedBrowsing`
* `UserAccess`

They aren't causing errors like `AmpPlugins` and `AmpThemes`, apparently, because either they are `Delayed` or because they appear alphabetically later in the list of `SERVICES` than `dependency_support`.

Reply:
It's strange that this is only happening at activation and not after having been activated.

Reply:
Error happens only during activation in the admin, not when activating via WP-CLI.

Reply:
I've temporarily changed the order of the `SERVICES`: c1e18f8a5.